src/modules/transaction/core/services/suarmi.service.new.ts

```
@@ -0,0 +1,171 @@
+      const res = await lastValueFrom(this.httpService.get(url, this.getHeaders()));
+      this.userRepository.save({ ...user, suarmi: res.data.uuid[0] });
+      return res.data.uuid[0];
+    } catch (error: any) {
+      throw new SuarmiUserVerificationError();
+    }
+  }
+
+  private async createUser(user: User): Promise<string> {
+    try {
+      const suarmiUserId = await this.suarmiUserExists(user);
+      if (suarmiUserId) return suarmiUserId;
+
+      const data = await this.formatSuarmiUserCreationData(user);
+
+      const url = `${this.suarmi.apiUrl}/user`;
+      const { uuid } = (await lastValueFrom(this.httpService.post(url, data, this.getHeaders()))).data;
+      await this.userRepository.save({ ...user, suarmi: uuid });
+      return uuid;
+    } catch (e: any) {
+      throw new SuarmiUserCreationError();
+    }
+  }
+
+  public async sendOrder(input: WithdrawUseCaseDto): Promise<{ txnInput: TransactionInput; data: Order }> {
+    const { type, bank_account, from_amount, to_amount, bank_name, userId: uid } = input;
+    const { to_currency } = await this.formatOrderInput(input);
+
+    if (type === SupportedTransactionType.WITHDRAW && !bank_account) throw new InvalidBankAccountError();
+
+    const user = await this.userRepository.findOneBy({ id: uid });
+    let { suarmi } = user;
+    const { walletAddress } = user;
+    if (suarmi === null) suarmi = await this.createUser(user);
+    const suarmiOrderInput = this.formatSuarmiOrder(input);
+    const url = `${this.suarmi.apiUrl}/order`;
+    const res = await lastValueFrom(
+      this.httpService
+        .post(
+          url,
+          {
+            ...suarmiOrderInput,
+            user: suarmi,
+            to_address: type === SupportedTransactionType.DEPOSIT ? walletAddress : bank_account,
+          },
+          this.getHeaders()
+        )
+        .pipe()
+    );
+    if (res.status != 201) throw new ApolloError('Error al crear orden de compra en Suarmi');
```

As suggestion, maybe we could also throw the error code for better debugging